src/PIL/Image.py

```diff
@@ -1145,6 +1145,11 @@ def crop(self, box=None):
         if box is None:
             return self.copy()
 
+        if box[2] < box[0]:
+            raise ValueError("Region right less than region left")
+        elif box[3] < box[1]:
+            raise ValueError("Region lower less than region upper")
```

Comment:
Looks good, maybe this is clearer?

```suggestion
        if box[2] < box[0]:
            raise ValueError("Coordinate 'right' is less than 'left'")
        elif box[3] < box[1]:
            raise ValueError("Coordinate 'lower' is less than 'upper'")
```